### PR TITLE
fix: Implement storage unit test[#45]

### LINKED ISSUE #45: implement storage unit test
refer to: https://github.com/dataswap/core/issues/60
Complete Exception Test Cases

### PR DESCRIPTION
**Pull request by lovel8**
## Related Issues

close #45 

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: test/v0.8/uinttests/module/storage/setup/StorageTestSetup.sol [hunk 1]
```diff
@@ -40,6 +40,7 @@ contract StorageTestSetup {
     Storages storages;
     StoragesAssertion assertion;
     StoragesHelpers helpers;
+    Generator generator = new Generator();
```
Comment (siriusyim) [inline test/v0.8/uinttests/module/storage/setup/StorageTestSetup.sol:43]: Recommend standardizing the style for initializing state variables

**Review by siriusyim — CHANGES_REQUESTED** [review]

File: test/v0.8/uinttests/module/storage/SubmitStorageDealId.t.sol [hunk 2]
```diff
@@ -17,7 +17,7 @@
 pragma solidity ^0.8.21;
 
 import {Test} from "forge-std/Test.sol";
-import {SubmitStorageDealIdTestCaseWithSuccess} from "test/v0.8/testcases/module/storage/SubmitStorageDealIdTestSuite.sol";
+import "test/v0.8/testcases/module/storage/SubmitStorageDealIdTestSuite.sol";
```
[the remark this answers is not in the document]
Comment (lovel8) [reply test/v0.8/uinttests/module/storage/SubmitStorageDealId.t.sol:20]: @siriusyim For situations where a large amount of symbol data needs to be imported, it is recommended to use this full import method uniformly, which will be neater

**Review by siriusyim — APPROVED** [review]

File: test/v0.8/testcases/module/storage/SubmitStorageDealIdTestSuite.sol [hunk 3]
```diff
@@ -56,3 +68,190 @@ contract SubmitStorageDealIdTestCaseWithSuccess is StoragesTestBase {
...
+        ) // solhint-disable-next-line
+    {}
+
+    function before() internal virtual override returns (uint64) {
+        (, uint64 matchingId) = storagesHelpers.setup();
+        return matchingId;
+    }
+
+    function action(uint64 _matchingId) internal virtual override {
+        //TODO: add getCars in matching:https://github.com/dataswap/core/issues/74
```
Comment (waynewyang) [inline test/v0.8/testcases/module/storage/SubmitStorageDealIdTestSuite.sol:98]: pls remove this TODO line.

File: test/v0.8/testcases/module/storage/SubmitStorageDealIdTestSuite.sol [hunk 4]
```diff
@@ -56,3 +68,190 @@ contract SubmitStorageDealIdTestCaseWithSuccess is StoragesTestBase {
...
+        (, bytes32[] memory cars /*uint64 size*/, , , ) = storages
+            .matchings()
+            .getMatchingTarget(_matchingId);
```
Comment (waynewyang) [inline test/v0.8/testcases/module/storage/SubmitStorageDealIdTestSuite.sol:101]: using `getMatchingCars`.

File: test/v0.8/testcases/module/storage/SubmitStorageDealIdTestSuite.sol [hunk 5]
```diff
@@ -56,3 +68,190 @@ contract SubmitStorageDealIdTestCaseWithSuccess is StoragesTestBase {
...
```
Comment (waynewyang) [inline test/v0.8/testcases/module/storage/SubmitStorageDealIdTestSuite.sol:185]: remove TODO, and using getMatchingCars.

File: test/v0.8/testcases/module/storage/SubmitStorageDealIdTestSuite.sol [hunk 6]
```diff
@@ -56,3 +68,190 @@ contract SubmitStorageDealIdTestCaseWithSuccess is StoragesTestBase {
...
```
Comment (waynewyang) [inline test/v0.8/testcases/module/storage/SubmitStorageDealIdTestSuite.sol:238]: remove TODO, and using getMatchingCars.

File: test/v0.8/uinttests/module/storage/SubmitStorageDealId.t.sol (hunk 2 above)
Comment (waynewyang) [inline test/v0.8/uinttests/module/storage/SubmitStorageDealId.t.sol:20]: pls using format: import {* ,* ,*} from .

**Review by waynewyang — CHANGES_REQUESTED** [review]

**Comment by lovel8** [comment]
@waynewyang Have done

**Review by waynewyang — APPROVED** [review]

**Pull request merged**